libarchive/archive_read_support_format_rar.c

```diff
@@ -3081,11 +3081,16 @@ copy_from_lzss_window(struct archive_read *a, void *buffer,
 
 static int
 copy_from_lzss_window_to_unp(struct archive_read *a, const void **buffer,
-                             int64_t startpos, int length)
+                             int64_t startpos, size_t length)
 {
   int windowoffs, firstpart;
   struct rar *rar = (struct rar *)(a->format->data);
 
+  if (length > rar->unp_buffer_size)
```

Comment:
This is part of the safety fix. Without it, we would still attempt to memcpy 18 quintillion bytes into `rar->unp_buffer` :)